pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/CreatePastebinWorkitemHandler.java

```diff
@@ -0,0 +1,153 @@
+import java.util.Map;
+
+import org.jbpm.document.Document;
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.RequiredParameterValidator;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.jbpm.process.workitem.core.util.service.WidAction;
+import org.jbpm.process.workitem.core.util.service.WidService;
+import org.jpastebin.pastebin.PastebinLink;
+import org.jpastebin.pastebin.PastebinPaste;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "CreatePastebin.wid", name = "CreatePastebin",
+        displayName = "CreatePastebin",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.pastebin.CreatePastebinWorkitemHandler()",
+        documentation = "${artifactId}/index.html",
+        parameters = {
+                @WidParameter(name = "Title"),
+                @WidParameter(name = "Content", required = true),
+                @WidParameter(name = "Format"),
+                @WidParameter(name = "Visibility"),
+                @WidParameter(name = "Author")
+        },
+        results = {
+                @WidResult(name = "PasteURL")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+        },
+        serviceInfo = @WidService(category = "${name}", description = "${description}",
+                keywords = "paste,pastebin,create",
+                action = @WidAction(title = "Create a paste on Pastebin")
+        ))
+public class CreatePastebinWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private String develKey;
+
+    private static final Logger logger = LoggerFactory
+            .getLogger(CreatePastebinWorkitemHandler.class);
+    private static final String RESULTS_VALUE = "PasteURL";
+
+    public CreatePastebinWorkitemHandler(String develKey) {
+        this.develKey = develKey;
+    }
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+        try {
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+
+            String title = (String) workItem.getParameter("Title");
+            Object content = workItem.getParameter("Content");
+            String format = (String) workItem.getParameter("Format");
+            String visibility = (String) workItem.getParameter("Visibility");
+            String author = (String) workItem.getParameter("Author");
+
+            // for testing
+            PastebinPaste testPastebin = (PastebinPaste) workItem.getParameter("TestPastebin");
```

Comment:
maybe replace this (and the below if else) with following

```
PastebinPaste pastebin = (PastebinPaste) workItem.getParameter("Pastebin");
if (pastebin == null) {
    pastebin = new PastebinPaste();
}
```

that way it's not really test specific as users can simply add their pastebins manually. Moreover you can even skip all other parameters if the pastebin is given...

Reply:
done